### DISCUSSION TITLE
Seemless experience for sign-up and sign-in process desired

### DISCUSSION THREAD

**Discussion by stx-chris** [clerk/javascript#3031] [Feature Requests]
We wanted to use Clerk for a seemless user experience in our Next.js applications. The need to differentiate between sign-up and sign-in is not only confusing to the user, but in our view also unwarranted. It is frustrating if you use sign-up, only to be told that you signed up before and have to use the sign-in form. To circumvent that, we ended up implementing our own logic but we would rather use a robust standard component.

Is it possible to expose only a single sign-"on" component that checks internally whether the user has signed up before and handles the sign-up or sign-in processes internally. It would then be up to the app to respond accordingly.

The sign-on flow would then be as follows:
1. User selects the sign-on method
2. Clerk checks whether user is set up
3. Clerk starts sign-up flow if user not set up and sign-in flow otherwise.

Looking forward to your thoughts.

**Comment by LekoArts** [comment]
Please upvote and leave your feedback on the existing feature request: https://feedback.clerk.com/roadmap/2c96310a-847e-4859-9a4e-13d4e82ccd5a
